packages/core/src/embeddings/OpenAIEmbedding.ts

```diff
@@ -82,8 +86,8 @@ export class OpenAIEmbedding extends BaseEmbedding {
     return data[0].embedding;
   }
 
-  async getTextEmbedding(text: string): Promise<number[]> {
-    return this.getOpenAIEmbedding(text);
+  async getTextEmbedding(texts: string | string[]): Promise<number[]> {
```

Comment:
looks great, but if `texts.length` > `embedBatchSize`, we might end up having more than `embedBatchSize` embeddings per request.

Reply:
> looks great, but if `texts.length` > `embedBatchSize`, we might end up having more than `embedBatchSize` embeddings per request.

the embedBatchSize is just for getTextEmbeddingBatch, it doesn't imply on how getTextEmbedding works, and I even think that it shouldn't because is not a max tokens or something like this

Reply:
I guess it all depends on the contract we want to give to the user.

I assumed that the user could use any embedding (OpenAI, Mixtral, etc.) with this contract:

```
getTextEmbedding(texts: string | string[]): Promise<Embedding>; getQueryEmbedding(query: string | string[]): Promise<Embedding>;
 ```
 
Using this contract, the user can send a single string or an array.
 
And the user can optionally set a batch size.

Our code takes care to split the text into batches of the batch size, iterates over the batches, and then either
1. sends a whole batch to the embedding API (as supported by OpenAI and other providers) or
2. iterates over the batch entries and sends one request per entry

As long as we don't implement 1. for each provider, we could fallback to 2. using some code in the base class

This protects our users from sending too much data in one request.

The contract above uses overloading; we could alternatively use the same contract as in Python:
```
getTextEmbedding(text: string): Promise<Embedding>; 
getQueryEmbedding(query: string): Promise<Embedding>;
getTextEmbeddingsBatch(texts: string[]): Promise<Array<Embedding>>;
getQueryEmbeddingsBatch(queries: string[]): Promise<Array<Embedding>>;
 ```
 
Personally, I would prefer the first contract as it's easier for the user. @himself65 @yisding, what are your thoughts?

Reply:
I think we can keep it for another PR, we have nodeParsers that already solve a lot of split problems, and currently, we will add more complexity without much return, since we will need to count tokens from node text to fill batches correctly, and I don't see much benefits on overloading as well yet, I think it makes harder for adding new implementations since you'll always need to handle 2 different other cases, and would need to change the implementation on all current embeddings

Reply:
Obs: agree we can change later about splitting the text as well if needed, but now changing all it would be much more effort, and I think currently it will already help a lot.